### PR TITLE
Add image optimization task

### PR DESCRIPTION
**Pull request by johncionci**
## Summary

* Add gulp-imagemin to build task to compress theme image files

screen shot 2018-06-21 at 12 47 39 pm

## Metadata

| Question | Answer |
|----------|--------|
| Documentation reflects changes? | n/a
| `CHANGELOG` reflects changes? | n/a
| Unit/Functional tests reflect changes? | n/a
| Risk level | Low
| Relevant links | n/a

### CODE REVIEW

**Comment by nathandentzau** [comment]
Hey @johncionci we need to talk through this more. The file glob for the images is opinionated (locking us into an `img` directory) and ambiguous (no files extensions) which may cause us issues. I think this has value but we'd need to talk through some things:

* Our gulp tasks should include both modules and themes.
* Are we establishing a convention to force all image assets for themes and modules to live in a `img` directory? This does against our current practice of recursively searching through modules and themes for assets to build.
* What image types should we support for optimization?
* How do we handle this with our build process? In its current form, it's replacing the unoptimized image with an optimized one. This goes against the convention we have for the build tasks that compile assets into files we don't track with version control.

**Comment by johncionci** [comment]
@kbeck303 Same here. Looking for your insights with this. We can chat when you have time.

File: docs/gulp.md [hunk 1]
```diff
@@ -77,4 +78,9 @@ The `fix` command runs the syntax fixers provided by the code linters used for
 validating SCSS and JavaScript (ES6). The command will try its _best_ to fix
 coding standard violations based on specific rulesets for each language.
 
+## Optimize command
+The `optimize` command will run through the theme images directory
+`web/themes/custom/**/img/*` and reduce file size for all `.gif` `.jpeg` `.png`
```
Comment (marcaddeo) [inline docs/gulp.md]: Comma's between each file extension

**Comment by johncionci** [comment]
@jhogue same here.

**Comment by johncionci** [comment]
@jhogue @kbeck303 I think this actually should be optional, the problem with optional is that it may never be run... Thoughts?

**Comment by jhogue** [comment]
Going to need to give me some background here. When would this run, because it seems like it will only work on files stored locally, not files uploaded by an author.

**Comment by johncionci** [comment]
Correct, this is only for "theme images" - well thats not exactly true, its for any folder we tell it to run on - but you are correct, this would not effect any images uploaded via Drupal. This is more for developer added images. The way that its currently configured it runs anytime Gulp is run, but then you would end up "optimizing these images every commit which I think is overkill. We should prolly just have it as an available task and used as needed.

File: docs/gulp.md (hunk 1 above)
Comment (kbeck303) [reply docs/gulp.md]: something to note this folder could be `img` or `images` depending on the project.

**Comment by kbeck303** [comment]
Client uploaded image optimization would need to be handled separately.

**Comment by johncionci** [comment]
This is no longer relevant as we have moved away from gulp.

**Pull request closed without merging**